terraso_backend/audit_logs/api.py

```diff
@@ -0,0 +1,72 @@
+from datetime import datetime
+from typing import List, NewType, Protocol, Tuple
+
+from . import models
+
+# KeyValue represents a key-value pair
+KeyValue = NewType('KeyValue', Tuple[str, any])
+
+
+class AuditLog(Protocol):
```

Comment:
question: ok i am starting to have enough context to ask questions about protocols, in the code in this PR what is the advantage of using `Protocol`? my very new & shaky python understanding is that the code in `services.py` would still type check regardless of the use of `Protocol`, is that wrong?

Reply:
I'm also new to Python, so I may be wrong, but as far as I understand, you use the protocol as a type in the function that may spect the service. The service has to implement the protocol currently inheriting just because it helps me outcompete, but I will remove the inheritance. But any external service that expects logs of the typing will be the Protocol itself, and if the service does not implement it, it will fail a type check.

Reply:
> here are some initial questions/comments, i'm still wrapping my head around python/django so let me know if they don't make sense

Maybe a call would be worth it on this subject. @shrouxm @david-code

Reply:
hmm ok maybe i will clone the PR and try to get the type checker running and tweak some things to validate my understanding. what command are you running to type check the code?